cmd/dagger/functions.go

```diff
@@ -556,8 +556,11 @@ func (fc *FuncCommand) addArgsForFunction(cmd *cobra.Command, cmdArgs []string,
 	}
 
 	for _, arg := range fn.Args {
-		_, err := arg.AddFlag(cmd.Flags())
-		if err != nil {
+		if err := arg.AddFlag(cmd.Flags()); err != nil {
+			var e *UnsupportedFlagError
+			if errors.As(err, &e) {
```

Comment:
shall we print something in the CLI output to warn the user why that function is not being displayed? I'm thinking that some might be confused if they're defining functions with unsupported flags.

Reply:
That was my plan, yes, as commented in DEV-3613:

> It may be confusing for some users to create a function that doesn't show. But maybe we can add a summary at the end like "Skipped 13 unsupported functions", even if only on debug or something.

However, when making the changes to `module.go` I got concerned about having to "leak" that information from those functions somehow in order to present later at `--help`, in `FuncCommand`. With missing arguments it's doable since the function itself isn't skipped.

Also concerned about polluting `--help` as another positive for this feature is limiting the amount of core functions that get listed in `--help`, although it would add one line, instead of one for each skipped. 

Neither of those are blockers if it's the right thing to do but I'm wondering if using debug logs could be enough (still a challenge for skipped functions though). Didn't come back to revisit this, wdyt?